Conform to new selectionClipRect method in UIAsyncTextInput

megangardner opened the pull request:
#### ac9787d8128e0d370de5e47325b09c858a0085bd
<pre>
Conform to new selectionClipRect method in UIAsyncTextInput
<a href="https://bugs.webkit.org/show_bug.cgi?id=264551">https://bugs.webkit.org/show_bug.cgi?id=264551</a>
<a href="https://rdar.apple.com/118217775">rdar://118217775</a>

Reviewed by Wenson Hsieh.

Continuing adoption of more parts of the UIAsyncTextInput protocol

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _selectionClipRectInternal]):
(-[WKContentView _selectionClipRect]):
(-[WKContentView selectionClipRect]):

Canonical link: <a href="https://commits.webkit.org/270544@main">https://commits.webkit.org/270544@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1d7e968f92ce64e572bb6bac93fa6a5a2969a17b

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/25745 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/4352 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/27029 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/27846 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/23582 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/6109 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/1787 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/23694 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/25995 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/3266 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/22191 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/28426 "Built successfully") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/2891 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/29226 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/23501 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/23515 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/27090 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/2906 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/1137 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/4289 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6183 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/3358 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/3222 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

rr-codes commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -12061,6 +12066,12 @@ - (void)selectTextForContextMenuWithLocationInView:(CGPoint)locationInView compl
 
+- (CGRect)selectionClipRect
+{
+    WTFLogAlways("<MMG> selectionClipRect - new path");
```
should probably be removed 😛

aprotyas reviewed (approved):
Minus the logging call!

whsieh commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -2612,6 +2612,11 @@ - (CGRect)_selectionClipRect
 
+- (CGRect)_selectionClipRect
+{
+    return [self _selectionClipRectInternal];
```
```suggestion
{
    RELEASE_ASSERT_ASYNC_TEXT_INTERACTIONS_DISABLED();

    return [self _selectionClipRectInternal];
```
We can add an assertion here as well.

whsieh reviewed (approved):
r=me with the WTFLogAlways removed (and a `RELEASE_ASSERT_ASYNC_TEXT_INTERACTIONS_DISABLED`)

megangardner committed:
Conform to new selectionClipRect method in UIAsyncTextInput
https://bugs.webkit.org/show_bug.cgi?id=264551
rdar://118217775

Reviewed by Wenson Hsieh.

Continuing adoption of more parts of the UIAsyncTextInput protocol

* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKContentView _selectionClipRectInternal]):
(-[WKContentView _selectionClipRect]):
(-[WKContentView selectionClipRect]):

Canonical link: https://commits.webkit.org/270544@main
